src/test/java/htsjdk/samtools/cram/structure/BlockTest.java

```diff
@@ -0,0 +1,145 @@
+
+        Assert.assertEquals(actual.getUncompressedContent(), expectedRaw);
+        Assert.assertEquals(actual.getUncompressedContentSize(), expectedRaw.length);
+        Assert.assertEquals(actual.getCompressedContent(), expectedCompressed);
+        Assert.assertEquals(actual.getCompressedContentSize(), expectedCompressed.length);
+    }
+
+    @Test
+    public void uncompressedTest() {
+        final byte[] testData = "TEST STRING".getBytes();
+
+        final Block fhBlock = Block.createRawFileHeaderBlock(testData);
+        contentCheck(fhBlock, testData, testData);
+
+        final Block chBlock = Block.createRawCompressionHeaderBlock(testData);
+        contentCheck(chBlock, testData, testData);
+
+        final Block shBlock = Block.createRawSliceHeaderBlock(testData);
+        contentCheck(shBlock, testData, testData);
+
+        final Block core = Block.createRawCoreDataBlock(testData);
+        contentCheck(core, testData, testData);
+    }
+
+    private Block roundTrip(final Block in, final Version version) {
+        byte[] written;
+        try (final ByteArrayOutputStream os = new ByteArrayOutputStream()) {
+            in.write(version.major, os);
+            written = os.toByteArray();
+        } catch (final IOException e) {
+            throw new RuntimeIOException(e);
+        }
+
+        try (final InputStream is = new ByteArrayInputStream(written)) {
+            return Block.read(version.major, is);
+        } catch (final IOException e) {
+            throw new RuntimeIOException(e);
+        }
+    }
+
+    @Test
+    public void testFileHeaderBlockRoundTrips() {
+        final byte[] testData = "TEST STRING".getBytes();
+
+        final Block fhBlock = Block.createRawFileHeaderBlock(testData);
+
+        final Block rtBlock2 = roundTrip(fhBlock, CramVersions.CRAM_v2_1);
+        contentCheck(rtBlock2, testData, testData);
+
+        final Block rtBlock3 = roundTrip(fhBlock, CramVersions.CRAM_v3);
+        contentCheck(rtBlock3, testData, testData);
+    }
+
+    @Test
+    public void testCompressionHeaderBlockRoundTrips() {
+        final byte[] testData = "TEST STRING".getBytes();
+
+        final Block chBlock = Block.createRawCompressionHeaderBlock(testData);
+
+        final Block rtBlock2 = roundTrip(chBlock, CramVersions.CRAM_v2_1);
+        contentCheck(rtBlock2, testData, testData);
+
+        final Block rtBlock3 = roundTrip(chBlock, CramVersions.CRAM_v3);
+        contentCheck(rtBlock3, testData, testData);
+    }
+
+    @Test
+    public void testSliceHeaderBlockRoundTrips() {
+        final byte[] testData = "TEST STRING".getBytes();
+
+        final Block shBlock = Block.createRawSliceHeaderBlock(testData);
+
+        final Block rtBlock2 = roundTrip(shBlock, CramVersions.CRAM_v2_1);
+        contentCheck(rtBlock2, testData, testData);
+
+        final Block rtBlock3 = roundTrip(shBlock, CramVersions.CRAM_v3);
+        contentCheck(rtBlock3, testData, testData);
+    }
```

Comment:
Factoring out a data provider with {version, testdata} entries would eliminate the redundancy and reduce a lot of these test methods to just 2 or 3 lines.

Reply:
ok